src/v/cluster/log_eviction_stm.cc

```diff
@@ -174,7 +175,7 @@ log_eviction_stm::sync_effective_start(model::timeout_clock::duration timeout) {
             co_return errc::timeout;
         }
     }
-    co_return effective_start_offset();
+    co_return model::next_offset(_delete_records_eviction_offset);
```

Comment:
It's not obvious why this line changes: the difference is that the call to `effective_start_offset()` takes account of the last snapshot offset, whereas this doesn't, right?

Maybe the last snapshot offset is applied elsewhere after these changes?